javascript/javascript1/week2/smart-ease2.js

```diff
@@ -0,0 +1,135 @@
+//Flight booking fullname function
+
+function getFullname (firstname, surname, useFormalName) {
+    if (useFormalName == true) {
+        return "Lord " + firstname + " " + surname;
+    }
+    else {
+        return firstname + " " + surname;
+    }
+}
+
+//two variables assigned to getFullname function.
+let fullname1 = getFullname("Math", "Anton", true);
+let fullname2= getFullname("Mia", "David");
+
+//Log out the two fullname variables.
+console.log(fullname1);
+console.log(fullname2);
+
+
+//What do we do if useFormalName is not given as an argument?
+//But what if the person is a woman? Describe how you would fix the getFullname so it also works for women
+//below u can see what i did in those cases.
+
+/*
+function getFullname(firstName, surName, useFormalName, gender) {
+    if (typeof firstName !== 'string' || firstName === '') {
+       throw new Error('First name must be a non-empty string');
+    }
+     if (typeof surName !== 'string' || surName === '') {
+        throw new Error('last name must be a non-empty string');
+     }
+     
+    if (useFormalName) {
+
+        //question here, how can i specify (male and female) instead of string, so If something else is written except(male and female), error message shows
```

Comment:
Maybe using enums https://masteringjs.io/tutorials/fundamentals/enum

Reply:
This would return the error `getFullname("Nina", "Smith", "female");`, do you know why?

Reply:
Cuz **useFormalName** and **gender** are related to each other, we cant just have one of them either both or nothing. so there should be forth object before "female",  and for example (that object can be always the check box on the website)

Reply:
How can you solve that?

Reply:
Here is my solution: //Flight booking fullname function

`function getFullname(firstname, surname, useFormalName, gender) {
    if (typeof firstname !== 'string' || firstname === '') {
       throw new Error('First name must be a non-empty string');
    }
    if (typeof surname !== 'string' || surname === '') {
        throw new Error('sur name must be a non-empty string');
    } 

    if (useFormalName) { 
        if (gender) {
            if (gender === "male"){
                return `Lord ${firstname} ${surname}`;
            }
            else if (gender === "female"){
                return `Lady ${firstname} ${surname}`;
            }
            else{
                throw new Error('Gender must be male or female')
            } 
        }
        else if (gender===''){
            throw new Error('Gender must be a non-empty string')
        }
        else{
            return `Lord ${firstname} ${surname}`;
        }
    }
    else if (useFormalName === '') {
            throw new Error('Formal name must be a non-empty string')        
    }
    else {
        return `${firstname} ${surname}`;
    }
}
console.log(getFullname("Jad", "Sam", "sas", "female"));//Log out 

//two variables assigned to getFullname function.
const fullname1 = getFullname("Benjamin", "Hughes", true);
const fullname2= getFullname("Benjamin", "Hughes", false);

//Log out the two fullname variables.
console.log(fullname1);
console.log(fullname2);`

Reply:
how does it looks? do i miss something? 
If not, i would really like to see how you would make it ( shorter, cleaner or maybe you have another way for doing this exercise (better way))